src/osd/PrimaryLogPG.cc

```diff
@@ -11178,6 +11178,8 @@ uint64_t PrimaryLogPG::recover_backfill(
         assert(pbi.begin == check);
 
         to_remove.push_back(boost::make_tuple(check, pbi.objects.begin()->second, bt));
+        // Give deletion ops a cost in backfill, removing objects is not free.
+        ++ops;
```

Comment:
Have you tested this?  If we don't get replies for the deletes are you sure we'll get woken up again?